### ISSUE TITLE
Basic churn analysis

### ISSUE THREAD

**Issue by mvanwyk** [Data-Simply/pyretailscience#8]
Provide the basic tools to understand the retailer's churn impact

Questions that should be answered:
- [x] How many days typically occur between a customer's transactions?
- [ ] How many transactions occur before a customer churns?
- [ ] What is my churn period? ie at what point is it likely that X% of customers won't return
- [ ] Is my churn problem getting better or worse?
- [ ] What is the impact of churn on my revenue/customer count/etc?

Out of scope for the basic analysis:
* Churn modeling
* Segment analysis of churners vs non-churners
* Gradual churn (a slow reduction of engagement) analysis

**Comment by sgeorgedata** [comment]
I've started to look at this problem. I generated some data and have been looking at it with the scope of doing some basic statistics to see what the data is like before going on to look at the actual statistics. 

It's worth remembering this is my first look at this sort of data so I might be looking at this completely incorrectly.... 

I determined the days until churn per customer and visualised as a histogram:

![daytillchurn](https://github.com/Data-Simply/pyretailscience/assets/159256011/2a921475-edea-4c9b-975c-60be33366197)

The graph has the following outcomes:

Standard Deviation: 115.89
Maximum Value: 644
Minimum Value: 16
Churn Rate: 100.0 %

It almost looks like an exponential fall off or some Ricean distribution (but maybe I'd need to make the steps smaller on the left hand side to see this)

If we look at the data statistically we get:
Mean Days Until Churn: 114.88343558282209
Median Days Until Churn: 69.0
Mode Days Until Churn: 28


I haven't saved any of the code to the repo yet as I was mostly using to explore the data.

My next step is to run the simulation multiple times - I'm thinking with a range of different conditions to build up some statistically understanding of the limitations and then built in items above.

**Comment by mvanwyk** [comment]
@sgeorgedata I've been working on this as well. I realized that people (myself included) really talk about two different kinds of churn as one. There's transactional churn, or how many people churn after a specific number of transactions. And there's period churn, or how many people bought two periods ago who didn't buy in the last period. The former is applicable for customers who aren't in a stable buying pattern, and the latter for those who are.

It will be interesting to compare methodologies as I'm mostly going from my past industry experience. But as we've discussed, just because it's been done that way in industry doesn't mean it's the best way 😄

![image](https://github.com/Data-Simply/pyretailscience/assets/2493311/03286b9d-52e7-4a92-991d-66386e76acbc)
